### ISSUE TITLE
Client modules using typescript should depend on the typescript module

### ISSUE THREAD

**Issue by murdos** [jhipster/jhipster-lite#10421]
Following a [recent discussion](https://github.com/jhipster/jhipster-lite/pull/10404#discussion_r1693936727) on the addition of new modules adding typescript utilities, it demonstrates that we currently have an issue regarding typescript related modules.

Let's pick the `optional-typescript` as an example: it requires the `typescript`. 
So if you have applied for example the `vue-core` module, you can't use it unless you also apply the `typescript` module which will generates conflicting npm configuration (e.g. setting up jest in addition to already setup vitest).

On the contrary the new modules, [ts-loader](https://github.com/jhipster/jhipster-lite/pull/10404) and [ts-pagination-domain](https://github.com/jhipster/jhipster-lite/pull/10405) only requires the `init` module, which is convenient for matching both use cases, but a bit wrong since code will not compile if you don't have already applied a module that'll setup typescript.

My suggestion is to rework the `typescript` module to make it use a modern setup (eslint, vitest), and have at least `vue-core` and `react-core` modules (angular could be nice too, but it has a very specific build tools so this might be a bit more complicated) depend on the `typescript` module.

@pascalgrimaud @qmonmert @DamnClin @renanfranca : WDYT?

Related: https://github.com/jhipster/jhipster-lite/issues/3979

**Comment by renanfranca** [comment]
I agree with this 💯 👌.  It will help new frontend devs 😁👍

**Comment by maznag** [comment]
we need to remove `"type": "module"` from package.json for angular case. 

`npm run test:coverage` fail with this configuration and we get this error : `An unhandled exception occurred: require is not defined`

[error] ReferenceError: require is not defined
    at file:///................/jest.conf.js:1:37
    at ModuleJobSync.runSync (node:internal/modules/esm/module_job:400:35)
    at ModuleLoader.importSyncForRequire (node:internal/modules/esm/loader:427:47)
    at loadESMFromCJS (node:internal/modules/cjs/loader:1561:24)
    at Module._compile (node:internal/modules/cjs/loader:1712:5)
    at Object..js (node:internal/modules/cjs/loader:1895:10)
    at Module.load (node:internal/modules/cjs/loader:1465:32)
    at Function._load (node:internal/modules/cjs/loader:1282:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)

**Comment by murdos** [comment]
> we need to remove `"type": "module"` from package.json for angular case.

@maznag : could you open a new dedicated issue, with the steps to reproduce it? currently the `angular` module doesn't depend on the `typescript` module, so `"type": "module"` is not added unless you've also applied the `typescript` module.

**Comment by maznag** [comment]
@murdos yes we have this error when we apply `angular` and `typescript` modules. here is the issue https://github.com/jhipster/jhipster-lite/issues/12524 with steps to reproduce
